apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -26,6 +29,57 @@ const getTypeForBlockId = blockId => {
 	return block ? block.name : null;
 };
 
+let trackerBlocksCounter = 0;
+
+/**
+ * This helper function trackers the given blocks recursively
+ * in order to be able to track also the inner ones.
+ * The event properties will be populated (optional) propertiesHandler function.
+ * It acts as a callback passing two current block and the parent block (if exists).
+ * Take this as an advantage to add other custom properties to the event.
+ * Also, it adds as default `inner_block`,
+ * and `parent_block_client_id` and `parent_block_name` properties
+ * if the block has a parent block.
+ *
+ * @param {Array}    blocks            Block instances object or an array of such objects
+ * @param {string}   eventName         Event name used to track.
+ * @param {Function} propertiesHandler Callback function to populate event properties
+ * @param {object}   parentBlock       parent block. optional.
+ * @returns {void}
+ */
+function trackBlocks( blocks, eventName, propertiesHandler = noop, parentBlock ) {
+	const castBlocks = castArray( blocks );
+	if ( ! castBlocks || ! castBlocks.length ) {
+		return;
+	}
+
+	castBlocks.forEach( block => {
+		setTimeout(
+			( _block, _parent ) => {
+				const eventProperties = {
+					...propertiesHandler( _block, _parent ),
+					inner_block: !! _parent,
+				};
+
+				if ( _parent ) {
+					eventProperties.parent_block_client_id = _parent.clientId;
+					eventProperties.parent_block_name = _parent.name;
+				}
```

Comment:
I was wondering about this. I think it's good to have this info.

Reply:
yeah, it's related whit the @marekhrabe comment

>  ... haven't done it immediately as I was wondering if other actions we are tracking should also be looked at recursively,...

Reply:
I'm not sure the client_id would be useful. These ids are an implementation detail of the block editor and they are not persistent across edit sessions so I don't think we could gain anything from having them. The block name though is a perfect idea!